### ISSUE TITLE
Bounding issues related to +-MB and sphere_sweep's internal bounding.

### ISSUE THREAD

**Issue by wfpokorny** [POV-Ray/povray#59]
(Other sphere_sweep issues: #147 #285 #287)

For further background see attached file and the thread at:

http://news.povray.org/povray.bugreports/thread/%3Cweb.5692847b8bcbbc2219f71b680%40news.povray.org%3E/
## My view as also posted to the above thread:

I was reminded of this issue the other day and have been doing some more investigation.

I spent a lot of time comparing 3.6 to 3.7 code given 3.6 works only to find nothing. I finally stumbled across the fact the bounding mechanism is always on in 3.7. In other words (-MB<n> / +MB<n>) doesn't function 3.7.

In 3.6 we need 3 shapes (I think) or more for the bounding mechanism to trigger and we have 1 so internal bounding is off.

Is this the -MB / +MB mechanism supposed to work in 3.7 ? We changed the default threshold in the docs to 25 so guessing yes. I could find no 3.6 -> 3.7 change notes related to it.

The issue behind is just another issue with the bounding computation as you thought Christoph, and it is sitting there in 3.6 too - if we turn bounding on with +MB1.

Basically the current mechanism multiplies the radius of each sphere by 2 and in this case here where the radius is small, but the range is a couple orders of magnitude larger, that 2x just isn't enough. This even though the curve is a pretty gentle cubic as curves go.

The multiplier should probably fudge upward from 2 by default if the range for the curve is large compared to the radius. Another more general option would be to add a new keyword to the sphere_sweep which lets the user optionally control the bounding radius multiplier?

[the.pov.txt](https://github.com/POV-Ray/povray/files/238516/the.pov.txt)

**Comment by wfpokorny** [comment]
Just double checked the newsgroup link above and my last post is NOT there as I indicated though it is showing up in the 100 most recent posts thread. Not sure what happened in the web interface. The complete thread does show up properly in my thunderbird news reader.

**Comment by c-lipka** [comment]
The post is there alright, currently being number 5 of 6 in the thread.

**Comment by wfpokorny** [comment]
Yep, on manual reload the post is there. Firefox didn't pick up the updated page - should have thought to hit reload.

**Comment by wfpokorny** [comment]
This issue tangled with github issue #147

**Comment by wfpokorny** [comment]
A portion of the bounding issues here are related to issues with the bounding threshold control in 3.7. See issue #180 and the proposed pull request #181.

**Comment by wfpokorny** [comment]
Fix for the bounding issue was merged into the master branch May 14, 2017. 

However, merging the pull request #288 (restore complete bounding control) is required before this can be closed.

**Comment by c-lipka** [comment]
@wfpokorny Can you review this and comment what the current status of the issue is?

**Comment by wfpokorny** [comment]
As I recall you fixed the bounding issues which uncovered that v3.7, v3.8 cannot completely turn off bounding with -MB. I've long been running with a fix for this in the povr branch. At least one newsgroup poster had to use my fix for some satellite rendering thing he was doing - while you were tuned out - to circumvent other bounding issues likely related to our bounding using floats and the large surface to orbital space dimensions he was using.   

Suppose my current changes something similar to pull #288 or the later pull #298. The latter in the collection of pull requests I got tired of re-basing on it becoming stagnant and I closed it myself. 

I'll just close this too. 

I'm OK in povr. If others at some point care with POV-Ray properl about the +-MB control not being completely off with -MB in v3.7/v3.8 like v3.6 and prior, they'll complain and it can be dealt with a new, clean-start, issue.
